Dart/src/com/jetbrains/lang/dart/ide/refactoring/ServerRefactoring.java

```diff
@@ -73,13 +73,16 @@ public void run(@NotNull ProgressIndicator indicator) {
     if (serverErrorStatus != null) {
       return serverErrorStatus;
     }
+    if (finalStatus == null) {
+      return null;
+    }
     RefactoringStatus result = new RefactoringStatus();
     result.merge(optionsStatus);
     result.merge(finalStatus);
     return result;
   }
 
-  @NotNull
+  @Nullable
   public RefactoringStatus checkInitialConditions() {
     ProgressManager.getInstance().run(new Task.Modal(null, "Checking initial conditions", true) {
```

Comment:
OK
I'm going to move the description into the dialog and put the refactoring name into the title.

![image](https://cloud.githubusercontent.com/assets/384794/9747548/0f1ace10-5633-11e5-8685-528d63d666c0.png)

![image](https://cloud.githubusercontent.com/assets/384794/9747568/2cbf1408-5633-11e5-92d8-150b220b6326.png)